### DISCUSSION TITLE
Why not just use Intent and State?

### DISCUSSION THREAD

**Discussion by king-ma1993** [arkivanov/MVIKotlin#84] [Q&A] [answered]
I don't know why transform Event to Intent,and transform State to Model,why not just use Intent and State,there are too many sealed class

**Comment by arkivanov** [comment, marked as answer]
Thanks for the question, concerted this to discussion.

That separation is to keep UI and Store decoupled. Also the separate Model may contain UI specific things, whereas State may contain business logic specific things. But this is not obligatory, you may use same classes if you like.

**Reply by king-ma1993** [reply]
Is there any schematic diagram of the architecture here, which can help to understand the architecture？

**Reply by arkivanov** [reply]
There are some diagrams in the docs:

1. https://arkivanov.github.io/MVIKotlin/
2. https://arkivanov.github.io/MVIKotlin/store.html

**Reply by king-ma1993** [reply]
What is the difference between MVI and Redux

**Comment by arkivanov** [comment]
MVI (Model-View-Intent) and Redux are both state management patterns.

Differences:

- MVI focuses on unidirectional data flow, meaning data flows in one direction from the model to the view, while updates to the model come from the view through intents. Redux also uses a unidirectional data flow but focuses on having a single source of truth for the state, stored in a store - typically one single store for the whole app.
- MVI is widely known in Android development, while Redux is widely known in React Native and web development.

MVIKotlin has a lot in common with Redux, but the most important difference is that it is recommended to have separate stores for different screens.

**Comment by king-ma1993** [comment]
There is no ViewModel and LiveData in IOS, where are the replace of them in cross platform?

**Reply by arkivanov** [reply]
Please refer to various [samples](https://github.com/arkivanov/MVIKotlin/tree/master/sample) and [documentation](https://arkivanov.github.io/MVIKotlin/). You don't need `ViewModel` on iOS, `Store` replaces `LiveData`.
